extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java

```diff
@@ -136,7 +137,31 @@ List<MessageBundleBuildItem> processBundles(BeanArchiveIndexBuildItem beanArchiv
                 ClassInfo bundleClass = bundleAnnotation.target().asClass();
                 if (Modifier.isInterface(bundleClass.flags())) {
                     AnnotationValue nameValue = bundleAnnotation.value();
-                    String name = nameValue != null ? nameValue.asString() : MessageBundle.DEFAULT_NAME;
+                    String name = nameValue != null ? nameValue.asString() : MessageBundle.DEFAULTED_NAME;
+                    if (name.equals(MessageBundle.DEFAULTED_NAME)) {
+                        if (bundleClass.nestingType() == NestingType.TOP_LEVEL) {
+                            name = MessageBundle.DEFAULT_NAME;
+                        } else {
+                            // The name starts with the DEFAULT_NAME followed by an undercore, followed by simple names of all
+                            // declaring classes in the hierarchy seperated by underscores
+                            List<String> enclosingNames = new ArrayList<>();
```

Comment:
This is absolutely fine, maybe you write it this way for clarity. Still why don't you build it from down to top like this? Reverse operation seems more expensive, though we are talking couple of names max
- (should prepend underscore? you can just check if StringBuilder is empty) Foo
- Bar_Foo
- msg_Bar_Foo

it is string + string, less expensive than reverse

Reply:
Hm, I'm not quite I understand but my comment in the code is definitely wrong - it should read `Foo declares Bar`, i.e. ...

```java
class Bar { // Bar is enclosing of Foo
  static class Foo { // Foo is enclosing of Baz
     @MessageBundle
     interface Baz { // Baz is ignored <-- we start here
     }
  }
}
```
And we do reverse the list because we start from the message bundle interface e.g. `Baz` and walk up the hierarchy.

> it is string + string, less expensive than reverse

This IMO does not really matter here ;-).

Reply:
I understand, thank you